### PR TITLE
Fix backward compatibility issue introduced by #126

### LINKED ISSUE #142: uninitialized constant ActiveRecord::SessionStore::Session
After updating to v1.1.2, I now receive the error `uninitialized constant ActiveRecord::SessionStore::Session`. I've figured out that this is caused by the change to how the `active_record/session_store/session` file is required (inside an Active Support `on_load` block), and that I'm using an initializer file for setting options.

I feel like that change to how files were required is a breaking change and the version should have been bumped to v2.0.0.

### PR DESCRIPTION
**Pull request by Edouard-chin**
- Fixes #142
-  The goal of #126 was to lazily require the 'session.rb' file as
  otherwise, we'd load `Active::Record` base too prematurely during
  the boot process.

  This change had the negative effect of breakin apps that were
  setting configuration on the `Session` **before** the hook was
  getting triggered, thus ending up with a "uninitialized constant"
  error.

  This patches move the `session_class` assignment when loading then
  `session.rb` file as well as autoloading the
  `AR::SessionStore::Session` constant so that application referencing
  it too early will still work.

### CODE REVIEW

File: lib/active_record/session_store.rb [hunk 1]
```diff
@@ -109,8 +111,9 @@ def self.dump(value)
   end
 end
 
-ActiveSupport.on_load(:active_record) do
-  require 'active_record/session_store/session'
+ActiveSupport.on_load(:active_record, yield: true) do
+  next unless ActiveRecord::SessionStore.autoload?(:Session)
+
   ActionDispatch::Session::ActiveRecordStore.session_class = ActiveRecord::SessionStore::Session
```
Comment (kaspth) [inline lib/active_record/session_store.rb]: Isn't this duplicated in the session class?

**Comment by Edouard-chin** [comment]
I agree it seem over complicated but couldn't find a better alternative.

 > Would it make more sense to extend the existing accessor seen in

I'm not sure if I understand what you mean, the problem will persist as soon as `ActiveRecord::SessionStore::Session` is referenced

**Comment by kaspth** [comment]
From https://github.com/rails/activerecord-session_store/issues/142#issuecomment-471585618:

```ruby
# config/initializers/session_store.rb
Rails.application.config.session_store(:active_record_store)

ActiveSupport.on_load(:active_record) do
  ActiveRecord::SessionStore::Session.serializer = :json
end
```

It's only referenced to set the serializer (unless I'm misunderstanding how the first line works).

**Comment by Edouard-chin** [comment]
Yeah that code is actually the right way to configure the session.

What we need to fix is the case where `ActiveRecord::SessionStore::Session` is called outside the `on_load(:active_record)` hook (like the doc advises to do https://github.com/rails/activerecord-session_store#configuration)

File: lib/active_record/session_store.rb (hunk 1 above)
Comment (rafaelfranca) [reply lib/active_record/session_store.rb]: Not sure if we should se the session_class when loading session. Do we need to do that?

Comment (Edouard-chin) [reply lib/active_record/session_store.rb]: We can't remove the session_class assignment when loading the `session.rb` file and just have the assignment in the lazy hook. This is because of the edge case I mentioned in the PR description.

TL;DR    If the host application reference `Session`, "session.rb" gets loaded, the `active_record` hook is dispatched, at this point the `Session` constant is not yet defined since the AS hook takes precedence, calling `Session` one more time in the hook will just raise a undefined constant error. Chicken and egg situation really

Comment (rafaelfranca) [reply lib/active_record/session_store.rb]: Ok, so we should just `require `'active_record/session_store/session'` in this hook them. If the file was already required it will just return `false`.

Comment (Edouard-chin) [reply lib/active_record/session_store.rb]: I have a gift for finding overcomplicated solutions 🤦‍♂️ . Thanks!

**Comment by Edouard-chin** [comment]
This should be good to go Rafael/Kasper

**Comment by sikachu** [comment]
Thank you! I'm going to push out `1.1.3`
